src/pages/FavouritesPage/FavouritesPage.tsx

```diff
@@ -1,30 +1,58 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
 import { useAppDispatch } from "../../hooks/useAppDispatch";
 import { useAppSelector } from "../../hooks/useAppSelector";
-import { deleteAllMovies } from "../../store/reducers/favouriteSlice";
+import {
+  deleteAllMovies,
+  serchMovie,
+  sortYearAscending,
+  sortYearDescending,
+} from "../../store/reducers/favouriteSlice";
 import "./style.css";
 
-import FavouritesMoviesComponent from "../../components/FavouritesMoviesComponent/FavouritesMoviesComponent";
+import FavouritesMoviesComponent from "../common/FavouritesMoviesComponent/FavouritesMoviesComponent";
+import { UpButton } from "../../components/UpButton/UpButton";
+import Categories from "../common/Categories/Categories";
+import { useDebounce } from "../../hooks/useDebounce";
 
 const FavouritesPage = () => {
   const dispatch = useAppDispatch();
   const { favouriteMovies } = useAppSelector(
     (state) => state.movieFavouriteReducer
   );
+  const [search, setSearch] = useState<string>("");
```

Comment:
`search` will type automatically because of provided default value "".  So u don't need to double type